catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryResultIterable.java

```diff
@@ -65,8 +68,7 @@ public QueryResultIterable(CatalogFramework catalog, Function<Integer, QueryRequ
      *                 the caller can control iteration based on their own logic
      * @param pageSize How many results should each page hold
      */
-    public QueryResultIterable(CatalogFramework catalog, Function<Integer, QueryRequest> filter,
-            int pageSize) {
+    public QueryResultIterable(CatalogFramework catalog, Filter filter, int pageSize) {
```

Comment:
Ahhhhh right understood. It's a bummer we can't implement that batching of results somehow under the covers of the actual `getResults` method, which is what i would have expected it do in the first place.